src/command_modules/azure-cli-network/azure/cli/command_modules/network/custom.py

```diff
@@ -487,6 +488,138 @@ def update_ag_redirect_configuration(cmd, instance, parent, item_name, redirect_
+
+
+def update_ag_rewrite_rule(instance, parent, cmd, rule_set_name, rule_name, sequence=None,
+                           request_headers=None, response_headers=None):
+    with cmd.update_context(instance) as c:
+        c.set_param('rule_sequence', sequence)
+        c.set_param('action_set.request_header_configurations', request_headers)
+        c.set_param('action_set.response_header_configurations', response_headers)
+    return parent
+
+
+def show_ag_rewrite_rule(cmd, resource_group_name, application_gateway_name, rule_set_name, rule_name):
+    client = network_client_factory(cmd.cli_ctx).application_gateways
+    gateway = client.get(resource_group_name, application_gateway_name)
+    return _find_child_item(gateway, rule_set_name, rule_name,
+                            path='rewrite_rule_sets.rewrite_rules', key_path='name.name')
+
+
+def list_ag_rewrite_rules(cmd, resource_group_name, application_gateway_name, rule_set_name):
+    client = network_client_factory(cmd.cli_ctx).application_gateways
+    gateway = client.get(resource_group_name, application_gateway_name)
+    return _find_child_collection(gateway, rule_set_name, path='rewrite_rule_sets.rewrite_rules', key_path='name')
+
+
+def delete_ag_rewrite_rule(cmd, resource_group_name, application_gateway_name, rule_set_name, rule_name, no_wait=None):
+    client = network_client_factory(cmd.cli_ctx).application_gateways
+    gateway = client.get(resource_group_name, application_gateway_name)
+    rule_set = _find_child_item(gateway, rule_set_name, path='rewrite_rule_sets', key_path='name')
+    rule = _find_child_item(rule_set, rule_name, path='rewrite_rules', key_path='name')
+    rule_set.rewrite_rules.remove(rule)
+    sdk_no_wait(no_wait, client.create_or_update, resource_group_name, application_gateway_name, gateway)
+
+
+def create_ag_rewrite_rule_condition(cmd, resource_group_name, application_gateway_name, rule_set_name, rule_name,
+                                     variable, no_wait=False, pattern=None, ignore_case=None, negate=None):
+    ApplicationGatewayRewriteRuleCondition = cmd.get_models(
+        'ApplicationGatewayRewriteRuleCondition')
+    ncf = network_client_factory(cmd.cli_ctx).application_gateways
+    ag = ncf.get(resource_group_name, application_gateway_name)
+    rule = _find_child_item(ag, rule_set_name, rule_name,
+                            path='rewrite_rule_sets.rewrite_rules', key_path='name.name')
+    new_condition = ApplicationGatewayRewriteRuleCondition(
+        variable=variable,
+        pattern=pattern,
+        ignore_case=ignore_case,
+        negate=negate
+    )
+    _upsert(rule, 'conditions', new_condition, 'variable')
+    if no_wait:
+        return sdk_no_wait(no_wait, ncf.create_or_update, resource_group_name, application_gateway_name, ag)
+    parent = sdk_no_wait(no_wait, ncf.create_or_update, resource_group_name, application_gateway_name, ag).result()
```

Comment:
Would LongRunningOperation be good here as well?

Reply:
Unnecessary. We simply poll the poller to get back the parent. I do see the other pattern for instantiating LROs directly, but this is probably more idiomatic. Those could likely be rewritten, but I don't plan to do that in this PR.

Reply:
Just saw this, sounds good.